src/output.rs

```diff
@@ -77,20 +77,24 @@ impl Container {
 impl Container {
     /// Rename all struct members to rust conventions
     pub fn rename(&mut self) {
-        for m in &mut self.members {
-            if self.is_enum {
-                let pascald = m.name.to_pascal_case();
-                if pascald != m.name {
-                    m.serde_annot.push(format!("rename = \"{}\"", m.name));
+        for (i, m) in self.members.iter_mut().enumerate() {
+            let new_name = if self.is_enum {
+                if m.name.parse::<u64>().is_ok() {
+                    // sanitize numeric enum variant names from golang
+                    format!("r#_{}", m.name)
+                } else if syn::parse_str::<syn::Ident>(&m.name).is_err() {
+                    // name is not a valid rust identifier -> generate one
```

Comment:
does this catch keywords? we currently have a hardcoded keyword list in main 🙃

Reply:
If it does, we need to move the keyword check in here first and handle that separately.

I think before we move on to a full Variant rename, we should see if underscore prefixing in a rawstring is a valid identifier (because that's going to be the majority case). If that's not the case, then we should do Variants (and we should probably do Variants for ALL members at that point because otherwise we'd have to check variant names against regexes and it feels brittle).

Reply:
> does this catch keywords? we currently have a hardcoded keyword list in main 🙃

Good point, I'll have to check. Is the plan to move all the renaming then from `main.rs` into the `Container::rename`? I think it would be best to have everything in one place.

> I think before we move on to a full Variant rename, we should see if underscore prefixing in a rawstring is a valid identifier (because that's going to be the majority case). 

Makes sense.

> If that's not the case, then we should do Variants (and we should probably do Variants for ALL members at that point because otherwise we'd have to check variant names against regexes and it feels brittle)

I don't understand what you mean here, if one Variant can't be prefixed with `r#_` use the generic name for all (even the ones that can be escaped with a raw string)?

Reply:
> keywords refactor and `Container::rename`

Yeah, agreed. if we are starting to cover keyword related things inside output.rs then we should factor all of that out of main if possible.

> if one Variant can't be prefixed with r#_ use the generic name for all (even the ones that can be escaped with a raw string)?

Yeah. Because otherwise you could end up with enums like:

```rust
enum Blah {
  SomeSmartName,
  SomeOtherName,
  Variant1
  SomeFourthName
}
```

and devs would likely goto upstream docs only to find that `Variant1` does not exist there and get confused. If we renamed ALL of them to `Variant{i}` then it's a little more clear that these types do not match directly to upstream and are more likely to read the `#serde(rename)]` attrs.

Reply:
This does make the loop a little more convoluted though... Maybe if we just used less generic names (like `VariantAlias{i}` or `KopiumAlias{i}` this wouldn't be a problem 🤔

Reply:
I think we should just go with `Kopium*` and assume there wont be any collisions, this require less logic in the generating code and should be safe enough to never have collisions (until we have a Kopium CRD).